components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1044,468 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
+                    sub_query.mark_wildcard_match_required();
+                } else if (false == var_has_wildcard && schema_type == "int"
+                           && EncodedVariableInterpreter::
+                                   convert_string_to_representable_integer_var(
+                                           raw_string,
+                                           encoded_var
+                                   ))
+                {
+                    sub_query.add_non_dict_var(encoded_var);
+                } else if (false == var_has_wildcard && schema_type == "float"
+                           && EncodedVariableInterpreter::convert_string_to_representable_float_var(
+                                   raw_string,
+                                   encoded_var
+                           ))
+                {
+                    sub_query.add_non_dict_var(encoded_var);
+                } else {
+                    auto& var_dict = archive.get_var_dictionary();
+                    if (var_has_wildcard) {
+                        // Find matches
+                        std::unordered_set<VariableDictionaryEntry const*> var_dict_entries;
+                        var_dict.get_entries_matching_wildcard_string(
+                                raw_string,
+                                ignore_case,
+                                var_dict_entries
+                        );
+                        if (var_dict_entries.empty()) {
+                            // Not in dictionary
+                            has_vars = false;
+                        } else {
+                            // Encode matches
+                            std::unordered_set<encoded_variable_t> encoded_vars;
+                            for (auto entry : var_dict_entries) {
+                                encoded_vars.insert(EncodedVariableInterpreter::encode_var_dict_id(
+                                        entry->get_id()
+                                ));
+                            }
+                            sub_query.add_imprecise_dict_var(encoded_vars, var_dict_entries);
+                        }
+                    } else {
+                        auto entry = var_dict.get_entry_matching_value(raw_string, ignore_case);
+                        if (nullptr == entry) {
+                            // Not in dictionary
+                            has_vars = false;
+                        } else {
+                            encoded_variable_t encoded_var
+                                    = EncodedVariableInterpreter::encode_var_dict_id(entry->get_id()
+                                    );
+                            sub_query.add_dict_var(encoded_var, entry);
+                        }
+                    }
+                }
+            }
+        }
+        if (false == has_vars) {
+            continue;
+        }
+        if (false == possible_logtype_entries.empty()) {
```

Comment:
It's not possible to be `false` at this point, is it?

Reply:
Right, cause it continues earlier
```
        if (possible_logtype_entries.empty()) {
            continue;
        }
        ```